core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/branch/ValidationCleanupService.java

```diff
@@ -0,0 +1,75 @@
+package com.b2international.snowowl.core.branch;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.core.runtime.jobs.ISchedulingRule;
+
+import com.b2international.snowowl.core.ServiceProvider;
+import com.b2international.snowowl.core.events.AsyncRequest;
+import com.b2international.snowowl.core.identity.User;
+import com.b2international.snowowl.core.jobs.JobRequests;
+import com.b2international.snowowl.core.jobs.SerializableSchedulingRule;
+import com.b2international.snowowl.core.validation.ValidationRequests;
+import com.google.common.base.Joiner;
+
+public class ValidationCleanupService {
+	
+	public static final SerializableSchedulingRule VALIDATION_ISSUE_CLEANUP_RULE = new SerializableSchedulingRule() {
+		
+		private static final long serialVersionUID = 1L;
+
+		@Override
+		public boolean isConflicting(ISchedulingRule rule) {
+			return rule == this;
+		}
+		
+		@Override
+		public boolean contains(ISchedulingRule rule) {
+			return rule == this;
+		}
+	};
+
+	public ValidationCleanupService() {
+	}
+	
+	public void removeStaleIssues(ServiceProvider context, String resourceURI) {
```

Comment:
Rename all these methods to `scheduleStaleIssueRemoval` since it is not actually removing anything from the index but schedules a job that removes it.